app/gui2/src/util/ast/index.ts

```diff
@@ -261,11 +400,131 @@ if (import.meta.vitest) {
     })
     for (const { ast, expected } of resultWithExpected) {
       expect(ast.type).toBe(expected?.type)
-      expect(readAstSpan(ast, code)).toBe(expected?.repr)
+      expect(readAstOrTokenSpan(ast, code)).toBe(expected?.repr)
     }
   })
 }
 
+/**
+ * AST with additional metadata containing AST IDs and original code reference. Can only be
+ * constructed by parsing a full module code.
+ */
```

Comment:
Right, this comment is actually invalid now, since the only public way of creating that structure is guaranteeing that the parsed AST is derived from provided source code. Therefore I should remove it.